File: tmp/fetch-jd.mjs
import https from 'https';
import http from 'http';

const url = process.argv[2];
if (!url) {
  console.error("Usage: node fetch-jd.mjs <url>");
  process.exit(1);
}

const lib = url.startsWith('https') ? https : http;
const req = lib.get(url, {
  headers: {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'
  }
}, (res) => {
  let data = '';
  res.on('data', (chunk) => data += chunk);
  res.on('end', () => console.log(data));
});

req.on('error', (e) => {
  console.error(e);
  process.exit(1);
});

File: tmp/fetch-jd.py
import sys
import urllib.request
import urllib.error

url = sys.argv[1]
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'})
try:
    with urllib.request.urlopen(req) as response:
        html = response.read().decode('utf-8', errors='ignore')
        with open('tmp/last_jd.html', 'w', encoding='utf-8') as f:
            f.write(html)
        print(html)
except urllib.error.URLError as e:
    print(f"Error fetching {url}: {e}", file=sys.stderr)
    sys.exit(1)
